### PR TITLE
sql: improve kill's description

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

improve kill's description.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: sql-statements/sql-statement-kill.md [hunk 1]
```diff
@@ -34,8 +34,9 @@ Query OK, 0 rows affected (0.00 sec)
 
 ## MySQL compatibility
 
-* By design, this statement is not compatible with MySQL by default. This helps prevent against a case of a connection being terminated on the wrong TiDB server, because it is common to place multiple TiDB servers behind a load balancer.
-* The `KILL TIDB` statement is a TiDB extension. If you are certain that the session you are attempting to kill is on the same TiDB server, set [`compatible-kill-query = true`](/tidb-configuration-file.md#compatible-kill-query) in your configuration file.
+* By design, `KILL` is not compatible with MySQL by default. This helps prevent against a case of a connection being terminated on the wrong TiDB server, because it is common to place multiple TiDB servers behind a load balancer.
+* You can use `KILL` statment If you are certain that the session you are attempting to kill is on the same TiDB server and set [`compatible-kill-query = true`](/tidb-configuration-file.md#compatible-kill-query) in your configuration file.
+* The `KILL TIDB` statement is a TiDB extension. You can kill the session without limitation.
```
Comment (TomShawn) [inline sql-statements/sql-statement-kill.md:39]:
```suggestion
* If you are certain that the session you are attempting to kill is on the same TiDB server, set [`compatible-kill-query = true`](/tidb-configuration-file.md#compatible-kill-query) in your configuration file.
* The `KILL TIDB` statement is a TiDB extension. You can kill the session without limitation.
```

**Review by TomShawn — APPROVED** [review]

File: sql-statements/sql-statement-kill.md [hunk 2]
```diff
@@ -34,8 +34,8 @@ Query OK, 0 rows affected (0.00 sec)
 
 ## MySQL compatibility
 
-* By design, this statement is not compatible with MySQL by default. This helps prevent against a case of a connection being terminated on the wrong TiDB server, because it is common to place multiple TiDB servers behind a load balancer.
-* The `KILL TIDB` statement is a TiDB extension. If you are certain that the session you are attempting to kill is on the same TiDB server, set [`compatible-kill-query = true`](/tidb-configuration-file.md#compatible-kill-query) in your configuration file.
+* By design, `KILL` is not compatible with MySQL by default. This helps prevent against a case of a connection being terminated on the wrong TiDB server, because it is common to place multiple TiDB servers behind a load balancer. If you are certain that the session you are attempting to kill is on the same TiDB server, set [`compatible-kill-query = true`](/tidb-configuration-file.md#compatible-kill-query) in your configuration file.
```
Comment (mjonss) [inline sql-statements/sql-statement-kill.md:37]: I don't think changing the configuration is a good way to handle a specific session's decision if it is safe to assume that the client will be connected to the same TiDB node, which is how I interpret the current text.

Maybe rephrase to something like:
Only change the configuration if you are certain that clients always will be connected to the same TiDB node, since the default mysql client on `ctrl-c` will open a new connection to kill the currently running command, and if there are proxies in between, the new connection might be routed to a different TiDB node possibly killing a different session.

Will `KILL [TIDB]` terminate the running query or terminate the connection? (see https://dev.mysql.com/doc/refman/8.0/en/kill.html `KILL [CONNECTION]` vs `KILL QUERY`)

File: sql-statements/sql-statement-kill.md [hunk 3]
```diff
@@ -34,8 +34,8 @@ Query OK, 0 rows affected (0.00 sec)
...
+* The `KILL TIDB` statement is a TiDB extension. You can kill the session without limitation.
```
Comment (mjonss) [inline sql-statements/sql-statement-kill.md:38]:
```suggestion
* The `KILL TIDB` statement is a TiDB extension. This is different syntax than the mysql `KILL [CONNECTION|QUERY]` command and the mysql command line `ctrl-c` feature, but is safe to use in the same TiDB node.
```

**Review by mjonss — COMMENTED** [review]
I think it can be a bit more clear on the reasons why `KILL` is different in TiDB vs MySQL and the reasons for `KILL TIDB` and the `compatible-kill-query` configuration.

File: sql-statements/sql-statement-kill.md (hunk 2 above)
Comment (hawkingrei) [reply sql-statements/sql-statement-kill.md:37]: I agree with you.

Comment (hawkingrei) [reply sql-statements/sql-statement-kill.md:37]: https://github.com/pingcap/tidb/blob/master/docs/design/2020-06-01-global-kill.md

I think The Global Kill is the final solution. but it is experimental.

File: sql-statements/sql-statement-kill.md [hunk 4]
```diff
@@ -34,8 +34,9 @@ Query OK, 0 rows affected (0.00 sec)
 
 ## MySQL compatibility
 
-* By design, this statement is not compatible with MySQL by default. This helps prevent against a case of a connection being terminated on the wrong TiDB server, because it is common to place multiple TiDB servers behind a load balancer.
-* The `KILL TIDB` statement is a TiDB extension. If you are certain that the session you are attempting to kill is on the same TiDB server, set [`compatible-kill-query = true`](/tidb-configuration-file.md#compatible-kill-query) in your configuration file.
+* By design, `KILL` is not compatible with MySQL by default. This helps prevent against a case of a connection being terminated on the wrong TiDB server, because it is common to place multiple TiDB servers behind a load balancer. If you are certain that the session you are attempting to kill is on the same TiDB server.
```
Comment (mjonss) [inline sql-statements/sql-statement-kill.md:37]:
```suggestion
* By design, `KILL` is not compatible with MySQL by default. This helps prevent against a case of a connection being terminated on the wrong TiDB server, because it is common to place multiple TiDB servers behind a load balancer.
```

**Review by mjonss — APPROVED** [review]
LGTM after removal of unneeded sentence.

File: sql-statements/sql-statement-kill.md (hunk 4 above)
Comment (morgo) [reply sql-statements/sql-statement-kill.md:37]: In future it might be good to link to load balancing as well. But it has not merged yet: https://github.com/pingcap/docs/pull/5555
